src/dagmc.cpp

```diff
@@ -279,39 +232,32 @@ void load_dagmc_geometry()
   }
 
   if (!graveyard) {
-    warning("No graveyard volume found in the DagMC model."
+    warning("No graveyard volume found in the DagMC model. "
             "This may result in lost particles and rapid simulation failure.");
   }
 
   // --- Surfaces ---
 
   // initialize surface objects
-  int n_surfaces = model::DAG->num_entities(2);
-
+  int n_surfaces = dagmc_instance_->num_entities(2);
   for (int i = 0; i < n_surfaces; i++) {
-    moab::EntityHandle surf_handle = model::DAG->entity_by_index(2, i+1);
+    moab::EntityHandle surf_handle = dagmc_instance_->entity_by_index(2, i+1);
 
     // set cell ids using global IDs
-    DAGSurface* s = new DAGSurface();
+    auto s = std::make_unique<DAGSurface>();
     s->dag_index_ = i+1;
-    s->id_ = model::DAG->id_by_index(2, s->dag_index_);
-    s->dagmc_ptr_ = model::DAG;
-
-    if (contains(settings::source_write_surf_id, s->id_)) {
-      s->surf_source_ = true;
-    }
+    s->id_ = adjust_geometry_ids_ ? next_surf_id++ : dagmc_instance_->id_by_index(2, i+1);
+    s->dagmc_ptr_ = dagmc_instance_;
 
     // set BCs
     std::string bc_value = DMD.get_surface_property("boundary", surf_handle);
     to_lower(bc_value);
     if (bc_value.empty() || bc_value == "transmit" || bc_value == "transmission") {
-      // Leave the bc_ a nullptr
+      // set to transmission by default (nullptr)
     } else if (bc_value == "vacuum") {
       s->bc_ = std::make_shared<VacuumBC>();
     } else if (bc_value == "reflective" || bc_value == "reflect" || bc_value == "reflecting") {
```

Comment:
is this sort of thing not handled by DMD? I thought it dealt with synonyms?

Reply:
I believe synonyms are for the property type (`mat:`, `surf:`, `rho:`) as opposed to the value. The logic here is matches the surface BC logic [here](https://github.com/openmc-dev/openmc/blob/4399645859ce5459828721f12fbcaf052a63653a/src/surface.cpp#L137). I thought it made sense for these to be consistent.